src/databricks/labs/ucx/source_code/linters/imports.py

```diff
@@ -211,10 +217,27 @@ def visit_call(self, node: Call):
             return
         is_append = func.attrname == "append"
         changed = node.args[0] if is_append else node.args[1]
-        if isinstance(changed, Const):
-            self.sys_path_changes.append(AbsolutePath(node, changed.value, is_append))
-        elif isinstance(changed, Call):
-            self._visit_relative_path(changed, is_append)
+        relative = False
+        if isinstance(changed, Call):
+            if not self._match_aliases(changed.func, ["os", "path", "abspath"]):
+                return
+            relative = True
+            changed = changed.args[0]
+        try:
+            all_inferred = changed.inferred()
+            for inferred in all_inferred:
+                self._visit_inferred(changed, inferred, relative, is_append)
```

Comment:
This will not raise an `InferenceError`, right? I prefer to move it out of the try

Reply:
That would require `all_inferred` to be available outside the try-except clause which would increase complexity